system_tests/program_test.go

```diff
@@ -424,8 +427,11 @@ func storageTest(t *testing.T, jit bool) {
 	validateBlocks(t, 2, jit, builder)
 
 	// Captures a block_input_<id>.json file for the block that included the
-	// storage write transaction.
-	recordBlock(t, receipt.BlockNumber.Uint64(), builder)
+	// storage write transaction. Include wasm targets necessary for arbitrator prover and jit binaries
+	flag.Parse()
+	if *validatorInputsWriterBaseDir != "" {
```

Comment:
Are we intentionally trying to make it so that omitting the validatorInputsWriterBaseDir flag is disabling the recording of a block? I'd sort of prefer to keep these controls separate (or just don't guard block recording at all.)
That is to say, if the test author includes block recording, then record the block every time the test is run.

If we're worried about cluttering up the disk or something, then, yeah, a flag like "--recordBlockInputs=true" might make sense. But, I don't think it should be complected with the flag that controls the baseDir.